install.sh

```diff
@@ -33,18 +33,18 @@ fi
 echo "Gateway configuration:"
 
 # Try to get gateway ID from MAC address
-# First try eth0, if that does not exist, try wlan0 (for RPi Zero)
-GATEWAY_EUI_NIC="eth0"
-if [[ `grep "$GATEWAY_EUI_NIC" /proc/net/dev` == "" ]]; then
-    GATEWAY_EUI_NIC="wlan0"
-fi
 
-if [[ `grep "$GATEWAY_EUI_NIC" /proc/net/dev` == "" ]]; then
+# Get first non-loopback network device that is currently connected
+NICS=$(ip link 2>&1 | grep "state UP" | grep -v LOOPBACK | sed -E 's/^[0-9]+: ([0-9a-z]+): .*/\1/')
+if [[ -z $NICS ]]; then
     echo "ERROR: No network interface found. Cannot set gateway ID."
     exit 1
 fi
+GATEWAY_EUI_NIC=$(echo $NICS | cut -d " " -f1)
 
-GATEWAY_EUI=$(ip link show $GATEWAY_EUI_NIC | awk '/ether/ {print $2}' | awk -F\: '{print $1$2$3"FFFE"$4$5$6}')
+# Then get EUI based on the MAC address of that device
+read GATEWAY_MAC </sys/class/net/$GATEWAY_EUI_NIC/address
```

Comment:
For consistency reasons, wouldn't it be better to use `GATEWAY_MAC=$(cat /sys/class/net/$GATEWAY_EUI_NIC/address)` instead?